pyvista/utilities/helpers.py

```diff
@@ -271,26 +271,29 @@ def vtk_points(points, deep=True):
                          f'Shape is {points.shape} and should be (X, 3)')
 
     # points must be contiguous
-    points = np.ascontiguousarray(points)
+    if not points.flags['C_CONTIGUOUS']:
+        points = np.ascontiguousarray(points)
```

Comment:
I figure you made this change because `ascontiguousarray` seems to copy subclasses (which I didn't know!).

I dug around a bit and found an alternative in the numpy API: [`np.require`](https://numpy.org/doc/stable/reference/generated/numpy.require.html)
```py
In [1]: import numpy as np, pyvista as pv
   ...: 
   ...: arr = pv.pyvista_ndarray(np.arange(5))
   ...: print(np.ascontiguousarray(arr) is arr)
   ...: print(np.require(arr, requirements=['C']) is arr)
False
True
```

Checking the flag ourselves is probably fine too (although in that case I'd use `points.copy('C')` for semantic reasons), just putting this option out there.

Reply:
That's way better.  Can you suggest that so you can get credit for it?

Reply:
Sure, thanks, although it's not about credit, just easier for you to add with some clicks :)

Reply:
> Sure, thanks, although it's not about credit, just easier for you to add with some clicks :)

I'm 100% not above admitting I'm lazy.  But hey, now you get credit too!